doc/api/net.md

```diff
@@ -515,56 +544,74 @@ changes:
     description: The `hints` option is supported now.
 -->
 
-Opens the connection for a given socket.
-
-For TCP sockets, `options` argument should be an object which specifies:
-
-  - `port`: Port the client should connect to (Required).
-
-  - `host`: Host the client should connect to. Defaults to `'localhost'`.
+* `options` {Object}
+* `connectListener` {Function} Common parameters of [`socket.connect()`][]
+  methods. Will be added as a listener for the [`'connect'`][] event once.
+* Returns: {net.Socket} The socket itself.
 
-  - `localAddress`: Local interface to bind to for network connections.
+Initiate the connection on a given socket.
 
-  - `localPort`: Local port to bind to for network connections.
+Note: this method is similar to a [connect(2)][] call, and before that,
```

Comment:
so, its because pointing to the system call man pages alone doesn't really explain things like what bind options were used that I don't think this is useful :-)

its common to say "bind to interface", because a machine's IPv4 unicast addresses are themselves bound to interfaces... so binding to an IP addr is pretty much "binding to an interface", you figure out what interface you want, figure out its address, then bind to the address.

it doesn't surprise me that if the address is raw (not a domain name) it doesn't get parsed to figure out the implied address family, but I haven't used net with IPv6 much, so I don't know how it works. Whether intended or not, how it works is how it should be documented!